compiler/il/OMRBlock.cpp

```diff
@@ -805,6 +807,687 @@ OMR::Block::splitWithGivenMethodSymbol(TR::ResolvedMethodSymbol *methodSymbol, T
+ * @param Node which requires to be uncommoned
+ * @param BitVector constaining information about the unavailable registers
+ * @return Pair of GlobalRegisterNumber which can be used to store node
+ */
+static std::pair<TR_GlobalRegisterNumber,TR_GlobalRegisterNumber> findAvailableRegister(TR::Compilation *comp, TR::Node *node, TR_BitVector &unavailableRegisters)
+   {
+   TR_GlobalRegisterNumber firstGPR = comp->cg()->getFirstGlobalGPR();
+   TR_GlobalRegisterNumber lastGPR = comp->cg()->getLastGlobalGPR();
+   TR_GlobalRegisterNumber firstFPR = comp->cg()->getFirstGlobalFPR();
+   TR_GlobalRegisterNumber lastFPR = comp->cg()->getLastGlobalFPR();
+   TR_GlobalRegisterNumber firstVRF = comp->cg()->getFirstGlobalVRF();
+   TR_GlobalRegisterNumber lastVRF = comp->cg()->getLastGlobalVRF();
+
+   std::pair<TR_GlobalRegisterNumber,TR_GlobalRegisterNumber> toReturn = std::make_pair<TR_GlobalRegisterNumber,TR_GlobalRegisterNumber>(-1,-1);
+
+   TR_GlobalRegisterNumber start, end;
+   TR::DataType dt = node->getType();
+   if (dt.isIntegral() || dt.isAddress())
+      {
+      start = firstGPR;
+      end = lastGPR;
+      }
+   else if (dt.isFloatingPoint())
+      {
+      start = firstFPR;
+      end = lastFPR;
+      }
+   else if (dt.isVector())
+      {
+      start = firstVRF;
+      end = lastVRF;
+      }
+   else
+      TR_ASSERT_FATAL(false, "Unknown data type encountered when trying to pick a register for postGRA block splitting!");
+
+   for (TR_GlobalRegisterNumber itr = start; itr < end; ++itr)
+      {
+      if (!unavailableRegisters.get(itr))
+         {
+         toReturn.first = itr;
+         unavailableRegisters.set(itr);
+         break;
+         }
+      }
+   if (node->requiresRegisterPair(comp) && toReturn.first > -1)
+      {
+      for (TR_GlobalRegisterNumber itr = start; itr < end; ++itr)
+         {
+         if (!unavailableRegisters.get(itr))
+            {
+            toReturn.second = itr;
+            unavailableRegisters.set(itr);
+            break;
+            }
+         }
+      if (toReturn.second == -1)
+         {
+         unavailableRegisters.reset(toReturn.first);
+         toReturn.first = -1;
+         }
+      }
+   return toReturn;
+   }
+
+/*
+ * Returns a boolean notifying if the registers used by given nodes are available or not
+ *
+ * @param Compilation object
+ * @param Node whose used registers are required to be checked for availability 
+ * @param BitVector constaining information about the unavailable registers
+ * @return A boolean notifying if the registers used by the RegStore can be uses for uncommoning node
+ */
+static bool checkIfRegisterIsAvailable(TR::Compilation *comp, TR::Node *node, TR_BitVector &unavailableRegisters)
+   {
+   // Following assert is to make sure that this function is called with node which have Global Registers associated with it.
+   TR_ASSERT_FATAL(node->getOpCode().isStoreReg(), "checkIfRegisterIsAvailable is intended to use with RegStore nodes only");
```

Comment:
Can we add the opcode we encountered here in the assert?

Reply:
Added in https://github.com/eclipse/omr/commit/e75062fc5ab9ebb3996cc60947cb12fb2f027537